hyphen.c

```diff
@@ -1076,17 +1076,20 @@ int hnj_hyphen_norm(const char *word, int word_size, char * hyphens,
 void hnj_hyphen_hyphword(const char * word, int l, const char * hyphens, 
     char * hyphword, char *** rep, int ** pos, int ** cut)
 {
-  int hyphenslen = l + 5;
+  /* hyphword buffer size must be at least 2 * l */
+  int hyphword_len = 2 * l - 1;
```

Comment:
Thanks for feedback!
You are right. 
Using unsigned type should be ok in this case I think. Something like:
` unsigned int hyphword_len = (l > 0) ? 2 * (unsigned int) l - 1 : 0;`

Yet I'll have a look at it.